src/main/java/org/terasology/launcher/packages/PackageManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.packages;
+
+import java.nio.file.Path;
+
+/**
+ * Handles installation, removal and update of game packages.
+ */
+public class PackageManager {
+    private final Storage onlineStorage;
+    private final LocalStorage localStorage;
+
+    public PackageManager(Path localDir) {
```

Comment:
A point of confusion for me: Are you using `LocalStorage` as another option for where the launcher can get packages from (so essentially a "local provide") _or_ as a _target_ for packages to be stored locally? Where's `localDir` intended to come from?

Reply:
For now, it is designed only to receive packages (so not a 'Provider') and to install/remove them from the filesystem. :slightly_smiling_face:  The `localDir` path (that points to local data dir) already gets loaded by the `LauncherInitTask`, and the package manager will just borrow it from there